### DISCUSSION TITLE
XDrip, Wear OS 3 and Android ... "apk file is not for wear os"

### DISCUSSION THREAD

**Discussion by HARDYQUICK** [NightscoutFoundation/xDrip#2668] [Q&A]
Hi,
Handy Samsung S22 Ultra, Android 13
Smartwatch Skagen Falster Gen 6

I've made a big mistake and made the upgrade to Wear OS 3

Now I have to connect the watch by the new Skagen-App. Connetcion by Wear OS won't work anymore.

But! XDrip+ can't be installed anymore! Sideload by "Wear Intaller 2" via ADB-Debug alwways says: "This apk is not for Wear OS"
I can be transfered to the watch and can be started one time, asking for "I agree of all terms", but then, XDrip will not run on the watch. There are no XDrip-watchfaces on the watch.

I tried several hints to downgrade Google Play Store on the Phone and "cancel Upgrades" on the Watch... nothing helps.

Is this a bug on XDrip, Wear OS or Skagen App?

Urgently need help!

Regards, Hardy

**Comment by psonnera** [comment]
Don't try to install xDrip+, but only the Wear extension. Before ADB try that https://xdrip.readthedocs.io/en/latest/smartwatch/wearinstall/#wear-os-2x-and-3x and with ADB, look at method 3: you need to extract only the wear extension from xDrip+. There also is a Wear Installer 2 that might do for you-

**Comment by HARDYQUICK** [comment]
Great! I will try Is Wear needed on the phone anymore? The phone is connected by the Skagen app... Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Patrick Sonnerat ***@***.***> Datum: 10.02.23  23:10  (GMT+01:00) An: NightscoutFoundation/xDrip ***@***.***> Cc: HARDYQUICK ***@***.***>, Author ***@***.***> Betreff: Re: [NightscoutFoundation/xDrip] XDrip, Wear OS 3 and Android ...
  "apk file is not for wear os" (Discussion #2668) 
Don't try to install xDrip+, but only the Wear extension. Before ADB try that https://xdrip.readthedocs.io/en/latest/smartwatch/wearinstall/#wear-os-2x-and-3x and with ADB, look at method 3: you need to extract only the wear extension from xDrip+. There also is a Wear Installer 2 that might do for you-

**Reply by psonnera** [reply]
I believe Wear is needed but since I haven't tried, you'll tell us...

**Comment by HARDYQUICK** [comment]
First of all.Xdrip is on the watch again! PuhThat's great Thank youI've removed Wear OS from the phone and nothing happened :-). Connection to XDrip and via Skagen App is still thereVon meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Patrick Sonnerat ***@***.***> Datum: 11.02.23  12:03  (GMT+01:00) An: NightscoutFoundation/xDrip ***@***.***> Cc: HARDYQUICK ***@***.***>, Author ***@***.***> Betreff: Re: [NightscoutFoundation/xDrip] XDrip, Wear OS 3 and Android ...
  "apk file is not for wear os" (Discussion #2668) 
I believe Wear is needed but since I haven't tried, you'll tell us...

**Comment by HARDYQUICK** [comment]
So it’s very tricky.
You don’t need top have Wear OS on the phone to run a Falster Gen 6.
But, in my case, if you want to run an older Falster 3 parallel (for backup), you need to have Wear OS… only for this old watch. The new Skagen App does not know anything About it’s own older products. But if you start Wear OS the first time, it will ask to install the Falster Gen6 first. Here you have to cancel the Installation, otherwise it will hang up or stuck inside the installation, so Wear OS can’t be used any more. After canceling installation of Falster Gen 6 you can proceed installation Falster 3. 
And don’t try to ask the Skagen „Hotline“ 😊 for this…. 



Gesendet von Mail für Windows

Von: Patrick Sonnerat
Gesendet: Samstag, 11. Februar 2023 12:03
An: NightscoutFoundation/xDrip
Cc: HARDYQUICK; Author
Betreff: Re: [NightscoutFoundation/xDrip] XDrip, Wear OS 3 and Android ..."apk file is not for wear os" (Discussion #2668)

I believe Wear is needed but since I haven't tried, you'll tell us...
